mpi4jax/_src/jax_compat.py

```diff
@@ -36,7 +25,42 @@ def check_jax_version():
     from jax.interpreters.xla import Token
 
 
+if jax_version >= versiontuple("0.3.15"):
+    # abstract eval needs to return effects
+    # see https://github.com/google/jax/issues/11620
+    from functools import wraps
+    from jax.interpreters import mlir
+    from jax._src.lax import control_flow as lcf
+
+    effect = object()
```

Comment:
I'd vote for making this a class/dataclass that can be serialised/deserialized, has a useful name and a static hash that does not change upon rerunning the code.
 
We don't currently support serialising jaxpr-s because the address of the MPI communicator changes, but it would be something we could support in the future, and making sure the hash is static should avoid complications down the line.

Reply:
I'm fine with making this a different type, but...

>  static hash that does not change upon rerunning the code

Isn't this impossible? At least in a way that avoids hash collisions?

Reply:
why would it be? You always risk having an hash collision anyway. I'm just proposing to keep a constant value.

Reply:
Guess I was mistaken, I thought `hash(object) == id(object)` which would have been unique. So fine with me.